terraform/phase-two/main.tf

```
@@ -183,3 +183,153 @@ resource "aws_scheduler_schedule" "c19_ajldka_long_term_etl_scheduler" {
     role_arn = aws_iam_role.c19_ajldka_long_term_etl_scheduler_role.arn 
   }
 }
+
+# ecs task execution iam role
+resource "aws_iam_role" "c19_ajldka_task_execution_role" {
+  name = "c19-ajldka-task-execution-role"
+  assume_role_policy = jsonencode({
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": "sts:AssumeRole",
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ecs-tasks.amazonaws.com"
+            }
+        }
+    ]
+})
+}
+
+# ecs task execution iam role policy
+resource "aws_iam_role_policy_attachment" "c19_ajldka_task_execution_policy" {
+  role= aws_iam_role.c19_ajldka_task_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+# ecs task definition
+resource "aws_ecs_task_definition" "c19_ajldka_ecs_plants_task_definition" {
+  family                   = "c19_ajldka_ecs_plants_task_definition"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = 256
+  memory                   = 512
+  execution_role_arn = aws_iam_role.c19_ajldka_task_execution_role.arn
+  task_role_arn = aws_iam_role.c19_ajldka_ecs_task_role.arn
+  container_definitions = jsonencode([{
+        name      = "c19-ajldka-plants-dashboard"
+        image     = "129033205317.dkr.ecr.eu-west-2.amazonaws.com/c19-ajldka-plants-dashboard:latest"
+        essential = true
+        portMappings = [
+        {
+          containerPort = 8501
+          hostPort      = 8501
+          protocol      = "tcp"
+        }
+        ],
+        environment = [
+        {name = "DB_HOST", value = var.DB_HOST},
+        {name = "DB_NAME", value = var.DB_NAME},
+        {name = "DB_USER", value = var.DB_USER},
+        {name = "DB_PASSWORD", value = var.DB_PASSWORD},
+        {name = "DB_PORT", value = var.DB_PORT},
+        {name = "DB_SCHEMA", value = var.DB_SCHEMA},
+        {name = "DB_DRIVER", value = var.DB_DRIVER},
+        {name = "AWS_ACCESS_KEY", value = var.AWS_ACCESS_KEY},
+        {name = "AWS_SECRET_ACCESS_KEY", value = var.AWS_SECRET_ACCESS_KEY},
```

Don't we need these to be
```suggestion
        {name = "AWS_ACCESS_KEY_AJLDKA", value = var.AWS_ACCESS_KEY},
        {name = "AWS_SECRET_ACCESS_KEY_AJLDKA", value = var.AWS_SECRET_ACCESS_KEY},
```
to avoid encountering errors?

---

You're right, thank you!!